### PR TITLE
Wind energy distance mask bugfix

### LINKED ISSUE #811: Wind Energy gives non sensical results when min_distance arg is non 0
Rob Griffin pointed out that the Wind Energy model outputs non logical values when the `Minimum distance for offshore wind farm installation` (`min_distance` arg_spec) value is non 0. I can confirm this using the model sample data:

``` json
{
    "args": {
        "aoi_vector_path": "../../../../Program Files/InVEST_3.10.1_x64/sample_data/WindEnergy/input/New_England_US_Aoi.shp",
        "avg_grid_distance": "4",
        "bathymetry_path": "../../../../Program Files/InVEST_3.10.1_x64/sample_data/Base_Data/global_dem.tif",
        "global_wind_parameters_path": "../../../../Program Files/InVEST_3.10.1_x64/sample_data/WindEnergy/input/global_wind_energy_parameters.csv",
        "land_polygon_vector_path": "../../../../Program Files/InVEST_3.10.1_x64/sample_data/Base_Data/global_polygon.shp",
        "max_depth": "60",
        "max_distance": "200000",
        "min_depth": "3",
        "min_distance": "100",
        "number_of_turbines": "80",
        "results_suffix": "mindist",
        "turbine_parameters_path": "../../../../Program Files/InVEST_3.10.1_x64/sample_data/WindEnergy/input/5_0_turbine.csv",
        "valuation_container": false,
        "wind_data_path": "../../../../Program Files/InVEST_3.10.1_x64/sample_data/WindEnergy/input/ECNA_EEZ_WEBPAR_Aug27_2012.csv"
    },
    "invest_version": "3.10.1",
    "model_name": "natcap.invest.wind_energy"
}
```

Rob reported values of 42, but I'm seeing outputs look like:

### PR DESCRIPTION
**Pull request by dcdenu4**
## Description
Wind energy had an inconsistency between what it expected from `_create_distance_raster`, a function that created a distance transform raster. The function docstring said it was creating a raster of euclidean distances but in reality it was using pixel distance. This caused some funky downstream issues including a distance masking bug among others. The function now calculates euclidean distances and updates downstream operations accordingly.

Also fixes a bug where a TaskGraph dependency was not handled properly, so running valuation in async mode caused an error.

Fixes #811 

## Checklist
- [x] Updated HISTORY.rst (if these changes are user-facing)
~~- [ ] Updated the user's guide (if needed)~~
- [x] Tested the affected models' UIs (if relevant)

### CODE REVIEW

File: src/natcap/invest/wind_energy.py [hunk 1]
```diff
@@ -970,7 +970,7 @@ def execute(args):
         dependent_task_list=[align_and_resize_density_and_harvest_task])
 
     LOGGER.info('Mask out depth and [distance] areas from Harvested raster')
-    task_graph.add_task(
+    masked_harvested_task = task_graph.add_task(
```
Comment (dcdenu4) [inline src/natcap/invest/wind_energy.py:973]: Need this handle to add as dependency in valuation component

File: src/natcap/invest/wind_energy.py [hunk 2]
```diff
@@ -1129,9 +1129,9 @@ def execute(args):
             if calc_grid_dist_without_land:
                 # Calculate distance raster without land points provided
                 final_dist_task = task_graph.add_task(
-                    func=_calculate_grid_dist_on_raster,
```
Comment (dcdenu4) [inline src/natcap/invest/wind_energy.py:1132]: `_calculate_grid_dist_on_raster` was essentially a copy of `_create_distance_raster` once the latter function used euclidean distances.

File: src/natcap/invest/wind_energy.py [hunk 3]
```diff
@@ -1734,7 +1734,7 @@ def _mask_by_distance(base_raster_path, min_dist, max_dist, out_nodata,
     """Create a raster whose pixel values are bound by min and max distances.
 
     Args:
-        base_raster_path (str): path to a raster with distance values.
+        base_raster_path (str): path to a raster with euclidean distance values.
```
Comment (emlys) [inline src/natcap/invest/wind_energy.py:1737]:
```suggestion
        base_raster_path (str): path to a raster with euclidean distance values in meters
```

File: tests/test_wind_energy.py [hunk 4]
```diff
@@ -589,6 +680,50 @@ def test_val_no_grid_land_pts(self):
...
+        args['n_workers'] = 1
+
+        wind_energy.execute(args)
+
+        raster_results = [
+            'carbon_emissions_tons.tif', 'levelized_cost_price_per_kWh.tif',
+            'npv.tif']
+
+        for raster_path in raster_results:
+            print(raster_path)
```
Comment (emlys) [inline tests/test_wind_energy.py:715]: Left over print statement?

**Review by emlys — CHANGES_REQUESTED** [review]
This all makes sense to me! Just had some very minor suggestions.

File: tests/test_wind_energy.py (hunk 4 above)
Comment (dcdenu4) [reply tests/test_wind_energy.py]: Thanks! Fixed in https://github.com/natcap/invest/pull/825/commits/af38235545341d721174ae57f75b296484a8f54a

File: src/natcap/invest/wind_energy.py [hunk 5]
```diff
@@ -1785,6 +1783,9 @@ def _create_distance_raster(base_raster_path, base_vector_path,
     LOGGER.info("Starting _create_distance_raster")
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply src/natcap/invest/wind_energy.py:1784]: Added in https://github.com/natcap/invest/pull/825/commits/af38235545341d721174ae57f75b296484a8f54a

**Comment by dcdenu4** [comment]
Thanks @emlys , back to you!

**Review by emlys — APPROVED** [review]
Thanks @dcdenu4!

**Pull request merged**